x-pack/plugins/apm/server/lib/services/get_service_metadata_details.ts

```diff
@@ -82,9 +80,7 @@ export async function getServiceMetadataDetails({
   ];
 
   const params = {
-    apm: {
-      events: [ProcessorEvent.transaction],
-    },
+    apm: { events: [ProcessorEvent.metric] },
```

Comment:
Pretty sure this should be conditional (either transaction or metric depending on the settings). 

We have a helper `getProcessorEventForAggregatedTransactions`:
https://github.com/elastic/kibana/blob/f4042dd56620c58a2ae84119443fda4ca735e21e/x-pack/plugins/apm/server/lib/service_map/get_service_map_service_node_info.ts#L129-L135

Reply:
> either transaction or metric depending on the settings

@sqren could you clarify a bit?

Reply:
Users can enable configure the APM Server to store transactions as (pre-aggregated) histogram metric docs. See more here: https://www.elastic.co/guide/en/apm/server/current/transaction-metrics.html

My point is: right now you assume they have it enabled. If the user changes back to transaction documents (`apm-server.aggregation.transactions.enabled: false`) this query will still read from metric index since it's hardcoded to that. 

Using `getProcessorEventForAggregatedTransactions` to determine the right event/index to query will solve this problem.

@dgieselaar Chime in if I skipped some important details.

Reply:
That sounds right. But why is this only querying transaction data?

Reply:
And do we really need all the should queries?

Reply:
I'd suggest profiling this query, you might be surprised 😅

Reply:
> And do we really need all the should queries?

I added the should to increase the score of the element which has these fields.

Reply:
@dgieselaar is `top metrics aggregation` the one I should use to replace the `shoulds` right?

Reply:
@cauemarcondes yes/maybe (I'm not sure if it works). And AFAIC you don't have to fix it right away, this information is not super critical to the user right away..